pkg/controller/migmigration/task.go

```diff
@@ -217,16 +222,15 @@ var FinalItinerary = Itinerary{
 		{Name: StageRestoreCreated, Step: StepStageRestore, all: HasStageBackup},
 		{Name: EnsureStagePodsDeleted, Step: StepStageRestore, all: HasStagePods},
 		{Name: EnsureStagePodsTerminated, Step: StepStageRestore, all: HasStagePods},
+		{Name: EnsureAnnotationsDeleted, Step: StepStageRestore, all: HasStageBackup},
```

Comment:
This seems like a reasonable move. Will probably speed the migration up a little bit to run this phase before waiting for Direct Migrations to complete.